p2p/host/basic/basic_host.go

```diff
@@ -500,13 +500,17 @@ func (h *BasicHost) makeUpdatedAddrEvent(prev, current []ma.Multiaddr) *event.Ev
 		return nil
 	}
 	prevmap := make(map[string]ma.Multiaddr, len(prev))
+	currmap := make(map[string]ma.Multiaddr, len(current))
```

Comment:
Do we even use `currmap`?

Reply:
I changed the iteration below to be over this map. This ensures that this method works even if there are duplicates in `Addrs` output.